### PR TITLE
updates for julia 1.0

### LINKED ISSUE #64: Testing fails on Julia 1.0
Just to bring this out in limelight
```
(v1.0) pkg> test ArrayViews
   Testing ArrayViews
    Status `/var/folders/j5/12tsmc_s7xs31bmt68r6z0xh4m4bb6/T/tmpoCm6qH/Manifest.toml`
  [a5c3d3bf] ArrayViews v0.7.0
  [34da2185] Compat v1.0.1
  [2a0f44e3] Base64  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Base64`]
  [ade2ca70] Dates  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Dates`]
  [8bb1440f] DelimitedFiles  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/DelimitedFiles`]
  [8ba89e20] Distributed  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Distributed`]
  [b77e0a4c] InteractiveUtils  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/InteractiveUtils`]
  [76f85450] LibGit2  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/LibGit2`]
  [8f399da3] Libdl  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Libdl`]
  [37e2e46d] LinearAlgebra  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/LinearAlgebra`]
  [56ddb016] Logging  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Logging`]
  [d6f4376e] Markdown  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Markdown`]
  [a63ad114] Mmap  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Mmap`]
  [44cfe95a] Pkg  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Pkg`]
  [de0858da] Printf  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Printf`]
  [3fa0cd96] REPL  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/REPL`]
  [9a3f8284] Random  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Random`]
  [ea8e919c] SHA  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/SHA`]
  [9e88b42a] Serialization  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Serialization`]
  [1a1011a3] SharedArrays  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/SharedArrays`]
  [6462fe0b] Sockets  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Sockets`]
  [2f01184e] SparseArrays  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/SparseArrays`]
  [10745b16] Statistics  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Statistics`]
  [8dfed614] Test  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Test`]
  [cf7118a7] UUIDs  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/UUIDs`]
  [4ec0a83e] Unicode  [`~/src/julia/usr/bin/../share/julia/stdlib/v1.0/Unicode`]
* running arrviews.jl ...
ERROR: LoadError: LoadError: UndefVarError: Range not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include at ./sysimg.jl:29 [inlined]
 [5] include(::String) at /Users/arora/.julia/packages/ArrayViews/SM75b/src/ArrayViews.jl:3
 [6] top-level scope at none:0
 [7] include at ./boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1038
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at ./boot.jl:319 [inlined]
 [12] eval(::Expr) at ./client.jl:389
 [13] top-level scope at ./none:3
in expression starting at /Users/arora/.julia/packages/ArrayViews/SM75b/src/common.jl:30
in expression starting at /Users/arora/.julia/packages/ArrayViews/SM75b/src/ArrayViews.jl:39
ERROR: LoadError: LoadError: Failed to precompile ArrayViews [a5c3d3bf-5697-5ba7-841a-c26aed266c5b] to /Users/arora/.julia/compiled/v1.0/ArrayViews/djLEy.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] include(::String) at ./client.jl:388
 [12] top-level scope at /Users/arora/.julia/packages/ArrayViews/SM75b/test/runtests.jl:10 [inlined]
 [13] top-level scope at ./none:0
 [14] include at ./boot.jl:317 [inlined]
 [15] include_relative(::Module, ::String) at ./loading.jl:1038
 [16] include(::Module, ::String) at ./sysimg.jl:29
 [17] include(::String) at ./client.jl:388
 [18] top-level scope at none:0
in expression starting at /Users/arora/.julia/packages/ArrayViews/SM75b/test/arrviews.jl:1
in expression starting at /Users/arora/.julia/packages/ArrayViews/SM75b/test/runtests.jl:7
ERROR: Package ArrayViews errored during testing
```

### PR DESCRIPTION
**Pull request by amontoison**
resolved #64

### CODE REVIEW

File: test/arrviews.jl [hunk 1]
```diff
@@ -1,7 +1,3 @@
-using ArrayViews
```
Comment (andreasnoack) [inline test/arrviews.jl]: Please don't remove these. They are convenient when running testing single files instead of the complete suite.

**Comment by Xx-fractale-xX** [comment]
Tests failed because the function `ContiguousView` is also present in `LinearAlgebra`.

**Comment by Xx-fractale-xX** [comment]
@andreasnoack, I commented two tests for information, I don't know where the problem come from. It's happen when we use a `stepped range`(e.g. a:2:b). Almost everything seem to work well with julia 1.0.

I also add an `appveyor.yml` file if you want to add `windows CI` in the future.

This package in very interesting for mathematics algorithm where it's frequent to extract matrix column.
The possibility to not allocated memory with `unsafe_aview` is also attractive (even if it's seem to be a problem for GC....).

I will use it for `Krylov` repository in `JuliaSmoothOptimizers` where we need to do lots of `view` for solving linear systems of high dimensions.

**Comment by VPetukhov** [comment]
Hi @andreasnoack,
Is there something, which prevents these fixes from merging? Looks like there are still issues with Julia 1.0

**Comment by VPetukhov** [comment]
Thanks!

**Comment by andreasnoack** [comment]
For this to really work, we'd need a transition to Project.toml instead of REQUIRE.

**Pull request merged**